src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApi.java

```diff
@@ -282,23 +282,40 @@ public CcdAuthenticator authenticateJurisdiction(String jurisdiction) {
         return authenticatorFactory.createForJurisdiction(jurisdiction);
     }
 
-    public StartEventResponse startEvent(
+    // ideally should be more generic. atm just used in creating new exception record as auto event
+    public CaseDetails createExceptionRecord(
         CcdAuthenticator authenticator,
         String jurisdiction,
         String caseTypeId,
-        String eventTypeId
+        String eventTypeId,
+        Function<StartEventResponse, CaseDataContent> caseDataContentBuilder,
+        String logContext
     ) {
```

Comment:
If it's `createExceptionRecord` then `eventTypeId` parameter should not be allowed.

So either rename the method or remove the param.

Reply:
`eventTypeId` is also used in `caseDataContentBuilder`. not sure which way it would be better to get rid of it right now :thinking:

Reply:
i suggest to leave it out as it is same thing for another auto action. generalising ccd api calls was the next step (as we discussed the other day). or if not next then just separate from all this refactor